scalajvm/app/controllers/Approving.scala

```diff
@@ -3,20 +3,23 @@ package controllers
 import helpers.{Notifications, ActionWithTx}
 import models.forms.{ApprovalForm}
 import models.queries._
-import play.api.Logger
-import play.api.Play.current
+import play.api.Logging
 import play.api.data.Forms._
 import play.api.data._
 import play.api.mvc._
+import play.api.i18n
+import play.api.Configuration
 import scalikejdbc._
 import security.BasicAuth
+import javax.inject._
 
 import scala.concurrent._
 import ExecutionContext.Implicits.global
 
-object Approving extends Controller {
+@Singleton
+class Approving @Inject()(implicit cc: ControllerComponents, configuration: Configuration) extends AbstractController(cc) with i18n.I18nSupport with Logging {
```

Comment:
Play! now uses dependency injection a lot, so I had to turn multiple `object`s into `@Singleton class`. That compiles without warnings and works, but I'm not sure I understand all the ramifications. Please tell me if there's any issues with such a substitution.